source/main/MainMenu.cpp

```diff
@@ -224,6 +224,11 @@ void MainMenu::MainMenuLoopUpdate(float seconds_since_last_frame)
             String terrain_name = Networking::GetTerrainName();
             if (terrain_name != "any")
             {
+                if (gui->GetMainSelector()->IsVisible())
```

Comment:
This looks like a workaround. We shouldn't show the selector window in the first place. I can't even figure out how could we end up in such state.

Reply:
`SkipMainMenu=Yes` combined with a server that enforces a certain terrain.

Reply:
The selector windows is opened before we know if the server enforces a certain terrain.

Reply:
Then we have 2 options:
* Fix `SkipMainMenu` to only show selector early in singleplayer, and to wait for server info in multiplayer. That's sensible design IMO.
* Add comment `// Config option [SkipMainMenu=Yes] opens the terrain selector before we know the server's map`

Reply:
You were right, we don't need to account for `SkipMainMenu` at all in multiplayer.

Reply:
And the clean code team won once again. :smile: